client/model.go

```diff
@@ -241,22 +241,36 @@ type TeamProjectAssignment struct {
 }
 
 type Policy struct {
-	Id                             string `json:"id"`
-	ProjectId                      string `json:"projectId"`
-	NumberOfEnvironments           int    `json:"numberOfEnvironments"`
-	NumberOfEnvironmentsPerProject int    `json:"numberOfEnvironmentsPerProject"`
-	RequiresApprovalDefault        bool   `json:"requiresApprovalDefault"`
-	IncludeCostEstimation          bool   `json:"includeCostEstimation"`
-	SkipApplyWhenPlanIsEmpty       bool   `json:"skipApplyWhenPlanIsEmpty"`
-	DisableDestroyEnvironments     bool   `json:"disableDestroyEnvironments"`
-	UpdatedBy                      string `json:"updatedBy"`
+	Id                         string `json:"id"`
+	ProjectId                  string `json:"projectId"`
+	NumberOfEnvironments       int    `json:"numberOfEnvironments"`
+	NumberOfEnvironmentsTotal  int    `json:"numberOfEnvironmentsTotal"`
+	RequiresApprovalDefault    bool   `json:"requiresApprovalDefault"`
+	IncludeCostEstimation      bool   `json:"includeCostEstimation"`
+	SkipApplyWhenPlanIsEmpty   bool   `json:"skipApplyWhenPlanIsEmpty"`
+	DisableDestroyEnvironments bool   `json:"disableDestroyEnvironments"`
+	SkipRedundantDepolyments   bool   `json:"skipRedundantDeployments"`
+	UpdatedBy                  string `json:"updatedBy"`
+}
+
+type PolicyCreatePayload struct {
+	ProjectId                  string `json:"projectId"`
+	NumberOfEnvironments       int    `json:"numberOfEnvironments,omitempty"`
+	NumberOfEnvironmentsTotal  int    `json:"numberOfEnvironmentsTotal,omitempty"`
+	RequiresApprovalDefault    bool   `json:"requiresApprovalDefault"`
+	IncludeCostEstimation      bool   `json:"includeCostEstimation"`
+	SkipApplyWhenPlanIsEmpty   bool   `json:"skipApplyWhenPlanIsEmpty"`
+	DisableDestroyEnvironments bool   `json:"disableDestroyEnvironments"`
+	SkipRedundantDepolyments   bool   `json:"skipRedundantDeployments"`
 }
 
 type PolicyUpdatePayload struct {
 	ProjectId                  string `json:"projectId"`
-	NumberOfEnvironments       int    `json:"numberOfEnvironments"`
+	NumberOfEnvironments       int    `json:"numberOfEnvironments,omitempty"`
+	NumberOfEnvironmentsTotal  int    `json:"numberOfEnvironmentsTotal,omitempty"`
```

Comment:
@bernot-dev I think that means we will not be able to disable the `NumberOfEnvironments` limit.  
In other words - if someone were to use the provider to set it, then would make a change in their terraform to remove it (`nil`) then it will _**not**_ be sent as part of the json payload, which mean it will not update its value at all.  

To update its value to `null` the backend expects one to explicitly tell it to `null` in the payload.  
Probably should have been a PATCH verb rather than PUT 🤷 

This is probably true for [these lines as well](https://github.com/env0/terraform-provider-env0/pull/151/files#diff-28a6f76d4db82dd4c73617ca57fa43c2eca0074f8b243cbe06aacb6311e2b168R258-R259)

Consider adding a test there to confirm the behavior I described